### ISSUE TITLE
Detect parameter already assigned via `functools.partial` and kwargs

### ISSUE THREAD

**Issue by aDotInTheVoid** [microsoft/pyright#9247]
**Is your feature request related to a problem? Please describe.**

When using `functools.partial`, it's possible to assign a parameter multiple times, by specifying it both as a keyword argument to the partial call, and then a regular argument to the later call.

Code sample in [pyright playground](https://pyright-play.net/?code=MQAgDgngTglg5gCwC4C4QGcmwMZIFB4wC2YA9lEiAGYCuAdrqaQDboF4AmApldUwBQBDNDDpIANCABGIsQEoQAWgB8IUajwgtIKFyQ0odEIJABqaeypMArCAC81eoxboAdGEEUYg5vyulJQTtrOQJ-a34pOwBGAAYFUAAFKFIpZi4iEDpSSlEQZlEuYylSADcuQPQQJAQYKuwfZjUqqkK8cP44hJAASSQAcg5pIroYbCKkUhA4PWMjLigUqBAEBa4gA)

```python
# pyright: strict

import functools


def foo(a: int, b: int) -> int:
    return a + b


foo5 = functools.partial(foo, a=5)

foo5(b=10) # Problem not in line above, as this call is fine
foo5(10) # It'd be nice to get an error here
```

This passes type checking, but then errors at runtime: 

```
% pyright ./mcve.py
0 errors, 0 warnings, 0 informations 
% python3.13 ./mcve.py 
Traceback (most recent call last):
  File "/Users/aloenr01/dev/feeds/./mcve.py", line 13, in <module>
    foo5(10)  # It'd be nice to get an error here
    ~~~~^^^^
TypeError: foo() got multiple values for argument 'a'
```

**Describe the solution you’d like**

Pyright should catch this and report "Parameter "a" is already assigned"

**VS Code extension or command-line**

pyright 1.1.385, CLI

**Comment by erictraut** [comment]
Thanks for the suggestion. This turns out to be a difficult feature to implement given the way that the `partial` support is implemented in pyright. It would end up being nearly a complete rewrite, and the resulting implementation wouldn't work as well with language server features like "signature help" (the component that displays the signature of a callable when filling in the arguments) because it's not possible to express that signature using normal Python callable forms. 

For those reasons, I'm going to reject this enhancement request at this time. I'm willing to revisit this in the future if there's sufficient demand (e.g. in the form of upvotes).
